reference-app/src/main/java/ch/admin/seco/service/reference/domain/occupation/AvamOccupation.java

```
@@ -195,6 +236,47 @@ public static AvamOccupation toAvamOccupation(AvamOccupationDto dto) {
             .build();
     }
 
+    boolean updatedLabel(AvamOccupationDto receivedDto) {
+        return !this.maleTextDe.equals(receivedDto.getMaleTextDe())
+            || !this.maleTextFr.equals(receivedDto.getMaleTextFr())
+            || !this.maleTextIt.equals(receivedDto.getMaleTextIt())
+            || !this.femaleTextDe.equals(receivedDto.getFemaleTextDe())
+            || !this.femaleTextFr.equals(receivedDto.getFemaleTextFr())
+            || !this.femaleTextIt.equals(receivedDto.getFemaleTextIt());
+    }
+
+    boolean expired(AvamOccupationDto receivedDto) {
+        if (this.validTo == null && receivedDto.getValidTo() == null) {
+            return false;
+        } else if (this.validTo == null && receivedDto.getValidTo() != null) {
+            return receivedDto.getValidTo().isBefore(TimeMachine.now().toLocalDate());
+        }
+        return this.validTo.isBefore(receivedDto.getValidTo());
```

I think this needs to be nullsafe

---

the only nullsafe which is not covered is the case when 
`this.validTo != null && receivedDto.getValidTo() == null` , 
which basically means we go from expired to _not_ expired.
I thought this case shouldn't be possible so I didn't "catch" it ...
Should I then throw an exception or should I trigger `AvamOccupationNotExpiredEvent`?

---

Yes, I think we can throw an Exception (at least until further tests) in that case, since I don't think the transition from expired to not expired should be possible. Maybe InvalidExpringDateTransition or something like that?